event-handlers/src/test/java/no/sikt/nva/nvi/events/evaluator/EvaluateNviCandidateHandlerTest.java

```diff
@@ -566,18 +603,269 @@ private static void mockSecretManager() {
+
+            assertEquals(testScenario.expectedEvaluatedMessage(), messageBody);
+        }
+
+        @Test
+        void shouldRejectUnverifiedAuthorsWithoutName() throws IOException {
+            var unnamedContributor = defaultUnverifiedContributor.withName(null);
+            verifiedContributors = emptyList();
+            unverifiedContributors = List.of(unnamedContributor);
+            var testScenario = getNonCandidateScenario();
+
+            handler.handleRequest(testScenario.event(), context);
+            var messageBody = getMessageBody();
+
+            assertEquals(testScenario.expectedEvaluatedMessage(), messageBody);
+        }
+
+        @Test
+        void shouldIdentifyCandidateWithBothVerifiedAndUnverifiedNviCreators() throws IOException {
+            unverifiedContributors = List.of(defaultUnverifiedContributor);
+            expectedTotalPoints = BigDecimal.valueOf(0.7071)
+                                            .setScale(SCALE, ROUNDING_MODE);
+            expectedPointsPerInstitution = List.of(new InstitutionPoints(CRISTIN_NVI_ORG_TOP_LEVEL_ID,
+                                                                         expectedTotalPoints,
+                                                                         List.of(new CreatorAffiliationPoints(
+                                                                             defaultVerifiedContributor.build()
+                                                                                                       .id(),
+                                                                             CRISTIN_NVI_ORG_SUB_UNIT_ID,
+                                                                             expectedTotalPoints))));
+            creatorShareCount = 2;
+            var testScenario = getCandidateScenario();
+
+            handler.handleRequest(testScenario.event(), context);
+            var messageBody = getMessageBody();
+
+            assertEquals(testScenario.expectedEvaluatedMessage(), messageBody);
+        }
+
+        @Test
+        void shouldIdentifyCandidateWithMissingCountryCode() throws IOException {
+            var affiliations = List.of(ExpandedAffiliation.builder()
+                                                          .withId(CRISTIN_NVI_ORG_SUB_UNIT_ID)
+                                                          .withCountryCode(null)
+                                                          .build());
+            var contributor = defaultVerifiedContributor.withAffiliations(affiliations);
+            verifiedContributors = List.of(contributor);
+            var testScenario = getCandidateScenario();
+
+            handler.handleRequest(testScenario.event(), context);
+            var messageBody = getMessageBody();
+
+            assertEquals(testScenario.expectedEvaluatedMessage(), messageBody);
+        }
+
+        @Test
+        void shouldRejectCandidateWithSwedishCountryCode() throws IOException {
```

Comment:
`...WithOnlySwedishCountryCode`? This can be read as swedish country code always results in `NonNviCandidate` 😄